cmd/kubeadm/app/phases/upgrade/postupgrade.go

```diff
@@ -160,14 +147,19 @@ func removeOldDNSDeploymentIfAnotherDNSIsUsed(cfg *kubeadmapi.MasterConfiguratio
 			installedDeploymentName = kubeadmconstants.CoreDNS
 			deploymentToDelete = kubeadmconstants.KubeDNS
 		}
-		dnsDeployment, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Get(installedDeploymentName, metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
-		if dnsDeployment.Status.ReadyReplicas == 0 {
-			return fmt.Errorf("the DNS deployment isn't ready yet")
+
+		// If we're dry-running, we don't need to wait for the new DNS addon to become ready
+		if !dryRun {
+			dnsDeployment, err := client.AppsV1().Deployments(metav1.NamespaceSystem).Get(installedDeploymentName, metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
+			if dnsDeployment.Status.ReadyReplicas == 0 {
+				return fmt.Errorf("the DNS deployment isn't ready yet")
+			}
 		}
-		err = apiclient.DeleteDeploymentForeground(client, metav1.NamespaceSystem, deploymentToDelete)
+
+		err := apiclient.DeleteDeploymentForeground(client, metav1.NamespaceSystem, deploymentToDelete)
```

Comment:
why is this not in the dry-run block?

Reply:
It shouldn't be, as the purpose of dryrun is to show what would be done. In this case, what would be done is that the following DNS deployment would get deleted. With the fake clientset, nothing happens though, this is just logged

Reply:
Ahh could you add a comment there then please, then lgtm